src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -252,6 +272,37 @@ bool convert_function_precision(
         }
     }
 
+    if (is_changed && !convert_input_output_precision) {
+        auto& results = f->get_results();
+        for (size_t i = 0; i < results.size(); i++) {
+            auto& result = results[i];
+            if (result->get_input_element_type(0) != orig_result_types[i]) {
+                auto result_input = result->input_value(0);
+                const auto convert = std::make_shared<ov::op::v0::Convert>(result_input, orig_result_types[i]);
+                if (result_input.get_node()->get_output_size() > 1) {
```

Comment:
is it possible to use op::util::create_ie_output_name function here?

Reply:
I'd prefer not to - if tensor associated with 'result_input' has legacy name set - create_ie_output_name will return it instead of this node friendly name. So I prefer to update node friendly name and and tensor legacy name separately.